connection/src/error.rs

```diff
@@ -42,6 +42,7 @@ impl Error {
         match self {
             Error::Grpc(_) => true,
             Error::Attestation(err) => err.should_retry(),
+            Error::TransactionValidation(ProposeTxResult::LedgerTxOutIndexOutOfBounds, _) => true,
```

Comment:
this makes the thick client attempt to retry if it encounters this new error code